scripts/MakeDepthPlot.R

```diff
@@ -1,124 +1,131 @@
-    }
-    maxDP = max(depth$DP)
-    depth$Nfract = depth$Ns / depth$DP * 100
-    depth$Nfract[depth$DP == 0] = 0
-    depth$Target = factor(namesVect, levels = c(myBed$Name,"Off_Target"))
-    maxNs = ceiling(max(depth$Ns[depth$Target != "Off_Target"])/3)*3
-    
-} else {
-    maxDP = 1
-    maxNs = 1
-    depth$Target = factor(namesVect, levels = c(myBed$Name,"Off_Target"))
-}
-multiplier = -maxDP / maxNs
-
-
-myFName = paste("Final/",inSampType,"/",inSampName, ".vcf", sep="")
-myVCF <- read_delim(
-  myFName, "\t", 
-  escape_double = FALSE, 
-  comment = "##", 
-  trim_ws = TRUE
-  )
-namesVect = c()
-if (length(row.names(myVCF)) > 0) {
-for (dataIter in seq(1,length(row.names(myVCF)))) {
-  myName = "Off_Target"
-  for (bedIter in seq(1, length(row.names(myBed)))) {
-    if ( myVCF$`#CHROM`[dataIter] == myBed$Chrom[bedIter] && 
-         myVCF$POS[dataIter] >= myBed$Start[bedIter] && 
-         myVCF$POS[dataIter] < myBed$End[bedIter]) {
-      myName = myBed$Name[bedIter]
-      break
-    }
-  }
-  namesVect = c(namesVect, myName)
-}
-myVCF$Target = factor(namesVect, levels = c(myBed$Name,"Off_Target"))
-} else {myVCF$Target = factor()}
-myPlot=ggplot(data = depth[depth$Target != "Off_Target",]) + 
-  geom_area(mapping = aes(x = Pos, y = DP), stat="identity") + 
-  geom_area(mapping=aes(x=Pos, y=Ns * multiplier), color='red', alpha=0.7) + 
-  geom_segment(data = myBed, mapping = aes(x=Start,xend=End, y=0, yend=0))
-if (length(row.names(myVCF)) > 0) {
-  myPlot = myPlot + 
-  geom_point(data=myVCF[myVCF$Target != "Off_Target",], mapping=aes(x=POS, y=maxDP), shape=1)
-}
-myPlot = myPlot + 
-  scale_y_continuous(
-    breaks = c(0, maxDP), 
-    sec.axis = sec_axis(~./multiplier, name="Ns (count)", breaks = c(0,maxNs/3, 2*maxNs/3,maxNs), labels = c(0,maxNs/3, 2*maxNs/3,maxNs))) + 
-  labs(title = inSampName) + 
-  theme_bw() + 
-  theme(
-    axis.title.y.right = element_text(color='red'), 
-    axis.text.x = element_text(angle=90)
-  ) + 
-  facet_wrap(. ~ Target, scales = "free_x", ncol = min(length(myBed$Name), 4))
-ggsave(filename = paste("Stats/plots/", inSampName, ".targetCoverage.png", sep=""), 
-       plot=myPlot, 
-       width = 200, 
-       height=50*ceiling(length(levels(depth$Target)) / 4), 
-       units="mm", 
-       device = "png")
+args = commandArgs(trailingOnly=TRUE)
+
+inBed = args[1]
+inSampName = args[2]
+inSampType = args[3]
+
+library(ggplot2)
+library(readr)
+# read in bed file
+# input:
+#        inBed = get_target_bed,
+#        inDepth = "{runPath}/Stats/data/{sample}.{sampType}.region.mutpos.vcf_depth.txt",
+
+#    output:
+#        "{runPath}/Stats/plots/{sample}.{sampType}.targetCoverage.png"
+
+myBed <- read_table2(inBed, col_names = FALSE)
+myFName = paste("Stats/data/",inSampName, ".region.mutpos.vcf_depth.txt", sep="")
+depth <- read_delim(myFName,
+                    "\t", escape_double = FALSE, trim_ws = TRUE, 
+                    col_types="cicii") 
+# Set column names

```

Comment:
I think this can be written better. It just seems to use a large number of 'else if' statements. Why not another approach such as (in R pseudocode)
x<-c("a", "b", "c","d")
colnames(x[:numCols])

Would that work?